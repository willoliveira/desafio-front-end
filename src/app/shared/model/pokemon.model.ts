import { IRessource } from "./ressource.model";

export enum EnumPokemonTypes {
	'normal',
	'fighting',
	'flying',
	'poison',
	'ground',
	'rock',
	'bug',
	'ghost',
	'steel',
	'fire',
	'water',
	'grass',
	'electric',
	'psychic',
	'ice',
	'dragon',
	'dark',
	'fairy',
	'unknown',
	'shadow'
}

export interface IPokemonAbilities {
	ability: IRessource;
	is_hidden: boolean;
	slot: number;
}

export interface IPokemonStats {
	base_stat: number;
	effort: number;
	stat: IRessource;
}

export interface IPokemonTypes {
	slot: number;
	type: IRessource;
}

export interface IPokemonSprites {
	back_default: string;
	back_female: string;
	back_shiny: string;
	back_shiny_female: string;
	front_default: string;
	front_female: string;
	front_shiny: string;
	front_shiny_female: string;
}

export interface IPokemon {
	id: number;
	base_experience: number;
	name: string;
	sprites: IPokemonSprites;
	weight: number;
	height: number;
	types: IPokemonTypes[];
	abilities: IPokemonAbilities[];
	stats: IPokemonStats[];
}
