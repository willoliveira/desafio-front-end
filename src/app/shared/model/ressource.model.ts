export interface IRessource {
	name: string;
	url: string;
}
