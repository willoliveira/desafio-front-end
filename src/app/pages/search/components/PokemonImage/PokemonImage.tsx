import React, { FC } from 'react';
import {  IPokemonSprites } from '../../../../shared/model/pokemon.model';
import { View, StyleSheet, Image } from 'react-native';

export interface IPokemonImageProps {
	sprites: IPokemonSprites;
	pokemonNumber: number;
}

export const pokemonCompanyImageUrl = 'https://assets.pokemon.com/assets/cms2/img/pokedex/full';
export const pokemonCompanyImageExt = 'png';

export const getPokemonCompanyImageUrl = (number: string): string => `${pokemonCompanyImageUrl}${number}.${pokemonCompanyImageExt}`

export const PokemonImage: FC<IPokemonImageProps> = ({ pokemonNumber, sprites }: IPokemonImageProps) => (
	<View style={styles.imageContainer}>
		<Image
			style={styles.image}
			source={{ uri: getUrl(pokemonNumber, sprites) }} />
	</View>
);

export const getUrl = (pokemonNumber: number, sprites: IPokemonSprites) => {
	let number = `${pokemonNumber}`
	if (pokemonNumber.toString().length === 1) {
		number = `00${number}`;
	} else if (pokemonNumber.toString().length === 2) {
		number = `0${number}`;
	}
	return getPokemonCompanyImageUrl(number);
	// return sprites && sprites.front_default;
}

const styles = StyleSheet.create({
	imageContainer: {
		backgroundColor: '#F2F2F2',
		borderRadius: 4,
		height: 300,
		width: '100%',
		padding: 24,
		flex: 1,
		alignContent: 'center',
		marginBottom: 12
	},
	image: {
		width: '100%',
		height: '100%'
	}
});
