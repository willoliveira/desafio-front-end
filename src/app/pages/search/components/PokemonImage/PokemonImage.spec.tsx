/**
 * @jest-environment jsdom
 */
import React from 'react';
import { shallow } from 'enzyme';
import { PokemonImage, getUrl, getPokemonCompanyImageUrl, pokemonCompanyImageExt, pokemonCompanyImageUrl } from './PokemonImage';
import { IPokemonSprites } from '../../../../shared/model/pokemon.model';

describe('<PokemonImage />', () => {

	let sprites: IPokemonSprites;

	beforeEach(() => {
		sprites = {
			"back_default": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/1.png",
			"back_female": null,
			"back_shiny": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/1.png",
			"back_shiny_female": null,
			"front_default": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png",
			"front_female": null,
			"front_shiny": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/1.png",
			"front_shiny_female": null
		};
	});

	describe('Rendering', () => {

		it('should match to snapshot', () => {
			const component = shallow(<PokemonImage sprites={sprites} pokemonNumber={1} />);
			expect(component).toMatchSnapshot("PokemonImage snapshot");
		});
	});

	describe('getPokemonCompanyImageUrl()', () => {
		it('Image pokemon number 1', () => {
			const number: string = '001';
			expect(getPokemonCompanyImageUrl('001')).toEqual(`${pokemonCompanyImageUrl}${number}.${pokemonCompanyImageExt}`);
		});

		it('Image pokemon number 25', () => {
			const number: string = '025';
			expect(getPokemonCompanyImageUrl('025')).toEqual(`${pokemonCompanyImageUrl}${number}.${pokemonCompanyImageExt}`);
		});

		it('Image pokemon number 125', () => {
			const number: string = '150';
			expect(getPokemonCompanyImageUrl('150')).toEqual(`${pokemonCompanyImageUrl}${number}.${pokemonCompanyImageExt}`);
		});
	})

	describe('getUrl()', () => {
		it('Image pokemon number 1', () => {
			expect(getUrl(1, sprites)).toEqual(getPokemonCompanyImageUrl('001'));
		});

		it('Image pokemon number 25', () => {
			expect(getUrl(25, sprites)).toEqual(getPokemonCompanyImageUrl('025'));
		});

		it('Image pokemon number 125', () => {
			expect(getUrl(150, sprites)).toEqual(getPokemonCompanyImageUrl('150'));
		});
	});
});
