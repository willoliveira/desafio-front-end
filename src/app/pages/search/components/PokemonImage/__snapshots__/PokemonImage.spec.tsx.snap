// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokemonImage /> Rendering should match to snapshot: PokemonImage snapshot 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "backgroundColor": "#F2F2F2",
      "borderRadius": 4,
      "flex": 1,
      "height": 300,
      "marginBottom": 12,
      "padding": 24,
      "width": "100%",
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://assets.pokemon.com/assets/cms2/img/pokedex/full001.png",
      }
    }
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  />
</View>
`;
